File: codes_5.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

vector<string> generateBinary(int n) {
    if (n == 0) {
        return {""};
    }
    
    vector<string> smaller = generateBinary(n - 1);
    vector<string> result;
    
    for (string s : smaller) {
        result.push_back(s + "0");
        result.push_back(s + "1");
    }
    
    return result;
}

int main() {
    int n = 3;
    vector<string> binaryStrings = generateBinary(n);
    
    cout << "Все бинарные строки длины " << n << ":" << endl;
    for (string s : binaryStrings) {
        cout << s << endl;
    }
    
    return 0;
}

File: codes_5.java
import java.util.ArrayList;
import java.util.List;

public class BinaryGenerator {
    public static List<String> generateBinary(int n) {
        if (n == 0) {
            List<String> result = new ArrayList<>();
            result.add("");
            return result;
        }
        
        List<String> smaller = generateBinary(n - 1);
        List<String> result = new ArrayList<>();
        
        for (String s : smaller) {
            result.add(s + "0");
            result.add(s + "1");
        }
        
        return result;
    }
    
    public static void main(String[] args) {
        int n = 3;
        List<String> binaryStrings = generateBinary(n);
        
        System.out.println("Все бинарные строки длины " + n + ":");
        for (String s : binaryStrings) {
            System.out.println(s);
        }
    }
}
